src/main/java/appeng/hooks/ticking/TickHandler.java

```diff
@@ -381,19 +376,19 @@ private void readyTiles(IWorld world) {
             // Chunks which are considered a border chunk will not "exist", but are loaded. Once this state changes they
             // will be readied.
             if (world.chunkExists(pos.x, pos.z) && chunkProvider.canTick(testBlockPos)) {
-                Queue<AEBaseTileEntity> queue = entry.getValue();
-
-                while (!queue.isEmpty()) {
-                    final AEBaseTileEntity bt = queue.poll();
+                Queue<AEBaseTileEntity> chunkQueue = worldQueue.remove(chunkPos);
+                if (chunkQueue == null) {
+                    continue; // This should never happen, chunk unloaded under our noses
+                }
 
+                for (AEBaseTileEntity bt : chunkQueue) {
                     // Only ready tile entites which weren't destroyed in the meantime.
                     if (!bt.isRemoved()) {
+                        // Note that this can load more chunks, but they'll at the earliest
+                        // be initialized on the next tick
                         bt.onReady();
                     }
                 }
-
-                // cleanup empty chunk queue
-                it.remove();
```

Comment:
Where is the processed queue for each separate chunk now removed or marked as done, so it will not run again the next tick?

Reply:
Commented on the remove() that it's removed and worked on completely